packages/protocol/contracts/ZkAsset/ZkAssetBurnable.sol

```diff
@@ -0,0 +1,47 @@
+
+
+pragma solidity >=0.5.0 <0.6.0;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+import "./ZkAsset.sol";
+import "../ACE/ACE.sol";
+import "../libs/LibEIP712.sol";
+import "../libs/ProofUtils.sol";
+
+contract ZkAssetBurnable is ZkAsset {
+    event UpdateTotalBurned(bytes32 noteHash, bytes noteData);
+
+    address public owner;
+
+    constructor(
+        address _aceAddress,
+        address _linkedTokenAddress,
+        uint256 _scalingFactor,
+        bool _canAdjustSupply,
+        bool _canConvert
+    ) public ZkAsset(
+        _aceAddress,
+        _linkedTokenAddress,
+        _scalingFactor,
+        _canAdjustSupply,
+        _canConvert
+    ) {
+    }
+
+    function confidentialBurn(uint24 _proof, bytes calldata _proofData) external {
+        require(_proofData.length != 0, "proof invalid");
```

Comment:
(This contract is just a template, but we should still put some permissioning on `mint` and `burn` methods. We should probably have this contract inherit from `Ownable.sol` and restrict this method to the owner only)

Reply:
Good idea, done